addons/hide-backpack/userscript.js

```diff
@@ -0,0 +1,86 @@
+export default async function ({ addon }) {
+  let originalBackpack = await getBackpackElement();
+  changeBackpackVisibility();
+
+  // Event listeners that add dynamic enable/disable + setting change
+  addon.settings.addEventListener("change", () => changeBackpackVisibility(addon));
+  addon.self.addEventListener("reenabled", () => changeBackpackVisibility(addon));
+  addon.self.addEventListener("disabled", () => {
+    moveResizeButtons(0);
+    originalBackpack.style.display = "block";
+    window.dispatchEvent(new Event("resize"));
+    document.querySelector(".sa-backpack-button").style.display = "none";
+  });
+
+  while (true) {
+    originalBackpack = await getBackpackElement();
+    if (!addon.self.disabled) changeBackpackVisibility();
+  }
+
+  async function getBackpackElement() {
+    return await addon.tab.waitForElement("[class^=backpack_backpack-header_]", {
+      markAsSeen: true,
+      reduxCondition: (state) => !state.scratchGui.mode.isPlayerOnly,
+    });
+  }
+
+  function changeBackpackVisibility() {
+    originalBackpack.style.display = "none";
+    let backpackEl = document.querySelector(".sa-backpack-button");
+    if (addon.settings.get("showButton")) {
+      if (backpackEl) {
+        moveResizeButtons(36);
+      } else {
+        createBackpackButton(addon);
+      }
+    } else {
+      if (isBackpackOpen()) originalBackpack.click();
+      moveResizeButtons(0);
+    }
+    document.body.classList.toggle("sa-backpack-open", isBackpackOpen());
+    window.dispatchEvent(new Event("resize"));
+  }
+}
+
+function isBackpackOpen() {
+  return !!document.querySelector("[class^=backpack_backpack-list_]");
+}
+
+// Create default backpack button
+function createBackpackButton(addon) {
```

Comment:
Is there a reason `addon` is being passed as a parameter?

Reply:
This function is defined outside the main function. It needs `addon` because it calls `addon.tab.scratchMessage()`.